src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -968,42 +968,47 @@ def create_x509_crl(self, builder, private_key, algorithm):
     def _create_x509_extensions(self, extensions, handlers, x509_obj,
                                 add_func, gc):
         for i, extension in enumerate(extensions):
-            try:
-                encode = handlers[extension.oid]
-            except KeyError:
-                raise NotImplementedError(
-                    'Extension not supported: {0}'.format(extension.oid)
+            if isinstance(extension.value, x509.UnrecognizedExtension):
+                x509_extension = self._create_unrecognized_x509_extension(
+                    extension
                 )
+            else:
+                try:
```

Comment:
Let's move everything in this else clause to a separate function to make this a bit more readable.